### PR TITLE
[KOGITO-3510] Release pipeline: set credentials as params

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-3510

Depends on:
- https://github.com/kiegroup/kogito-runtimes/pull/803
- https://github.com/kiegroup/kogito-images/pull/251
- https://github.com/kiegroup/kogito-cloud-operator/pull/595

### CODE REVIEW

**Review by sutaakar — COMMENTED** [review]
Looks good to me

File: Jenkinsfile.release [hunk 1]
```diff
@@ -53,17 +53,21 @@ pipeline {
...
         // IMAGE_PROMOTE_TAG => will be set in Initialization phase
 
-        AUTHOR_CREDS_ID = 'kie-ci'
-        GIT_AUTHOR='kiegroup'
+        // Credentials
+        GIT_AUTHOR='radtriste'
+        GIT_AUTHOR_CREDS_ID = 'radtriste' // username/password creds
+        BOT_AUTHOR='radtriste-bot-account'
+        BOT_AUTHOR_CREDS_ID='radtriste-bot' // username/password creds
+        GITHUB_TOKEN_CREDS_ID='radtriste-gh-token' // secret text creds
```
Comment (Kevin-Mok) [inline Jenkinsfile.release]: Should these be KIE credentials?

File: Jenkinsfile.release [hunk 2]
```diff
@@ -53,17 +53,21 @@ pipeline {
         // RELEASE_BRANCH => will be set in Initialization phase
 
         // We use quay as temporary registry for testing between the jobs instead of Openshift, due to https://issues.redhat.com/browse/KOGITO-2219
-        IMAGE_REGISTRY_CREDENTIALS = 'release_kogito'
+        IMAGE_REGISTRY_CREDENTIALS = 'tradisso_registry'
         IMAGE_REGISTRY = 'quay.io'
-        IMAGE_NAMESPACE = 'kiegroup'
+        IMAGE_NAMESPACE = 'tradisso'
```
Comment (Kevin-Mok) [inline Jenkinsfile.release]: Should these be reverted as well?

File: Jenkinsfile.release [hunk 3]
```diff
@@ -225,6 +241,11 @@ pipeline {
                     addStringParam(buildParams, 'DEPLOY_BUILD_URL', getJobUrl(IMAGES_DEPLOY))
                     addImageBuildParams(buildParams, 'PROMOTE', env.IMAGE_PROMOTE_TAG, true, true)
                     addBooleanParam(buildParams, 'DEPLOY_WITH_LATEST_TAG', params.DEPLOY_AS_LATEST)
+                    
+                    // To be uncommented for testing if artifacts are released to same repository as the staging
+                    addStringParam(buildParams, 'MAVEN_ARTIFACT_REPOSITORY', env.STAGING_REPOSITORY) // Get artifacts from staging repository
```
Comment (Kevin-Mok) [inline Jenkinsfile.release]: Should this be re-commented?

**Review by Kevin-Mok — CHANGES_REQUESTED** [review]
Seems like some lines need to be reverted from testing back to default, but it looks good otherwise. Great addition. :+1:

**Comment by radtriste** [comment]
@Kevin-Mok yeah I am still in draft mode in that one. I will just need to remove this commit: https://github.com/kiegroup/kogito-pipelines/pull/68/commits/95de24f2608edfb38687d8caa9d5aa31d991a6f3 ;)

File: README.md [hunk 4]
```diff
@@ -75,3 +75,60 @@ Here is the list of jobs and link to Jenkinsfiles:
...
+
+You will need to create single pipeline jobs and let them run once to update the `parameters` part (you should stop them quickly as it makes no sense to let them run until the end. Just wait for the checkout of repo and the `node` command done).
+
+**NOTE:** You will need to access the correct branch for each !
+
+* [kogito-release](./Jenkinsfile.release)
+* [create-release-branches](./Jenkinsfile.create-release-branches)
+* [kogito-runtimes-deploy](https://github.com/kiegroup/kogito-runtimes/blob/master/Jenkinsfile.deploy)
+* [kogito-runtimes-promote](https://github.com/kiegroup/kogito-runtimes/blob/master/Jenkinsfile.promote)
+* [[kogito-images-deploy](https://github.com/kiegroup/kogito-images/blob/master/Jenkinsfile.deploy)
```
Comment (Kevin-Mok) [inline README.md]:
```suggestion
* [kogito-images-deploy](https://github.com/kiegroup/kogito-images/blob/master/Jenkinsfile.deploy)
```

File: README.md [hunk 5]
```diff
@@ -75,3 +75,60 @@ Here is the list of jobs and link to Jenkinsfiles:
...
+
+* Have a specific author repository that you can test against
+* If you don't want to flood your main repository, you should use a "bot account", referred as `BOT_*`
+* For deploying runtimes artifacts, you will need to provide a nexus repository to deploy to (see [nexus-operator](https://github.com/m88i/nexus-operator))
+
+### Change pipeline envs
+
+* **GIT_AUTHOR** (and `GIT_AUTHOR_CREDS_ID`, see [Setup Jenkins creds](#setup-jenkins-creds))
+* **BOT_AUTHOR** (and `BOT_AUTHOR_CREDS_ID`, see [Setup Jenkins creds](#setup-jenkins-creds))
+* **IMAGE_NAMESPACE** (and `IMAGE_REGISTRY_CREDENTIALS`, see [Setup Jenkins creds](#setup-jenkins-creds))
```
Comment (Kevin-Mok) [inline README.md]:
```suggestion
* `GIT_AUTHOR` (and `GIT_AUTHOR_CREDS_ID`, see [Setup Jenkins creds](#setup-jenkins-creds))
* `BOT_AUTHOR` (and `BOT_AUTHOR_CREDS_ID`, see [Setup Jenkins creds](#setup-jenkins-creds))
* `IMAGE_NAMESPACE` (and `IMAGE_REGISTRY_CREDENTIALS`, see [Setup Jenkins creds](#setup-jenkins-creds))
```
I think monospace for variables is more standard?

File: README.md [hunk 6]
```diff
@@ -75,3 +75,60 @@ Here is the list of jobs and link to Jenkinsfiles:
...
+* GIT_AUTHOR_CREDS_ID (username/password credential)  
+  Username / [GitHub token](https://github.com/settings/tokens) which has rights on `GIT_AUTHOR`
+* BOT_AUTHOR_CREDS_ID (username/password credential)  
+  Username / [GitHub token](https://github.com/settings/tokens) which has rights on `BOT_AUTHOR`
+* GITHUB_TOKEN_CREDS_ID (secret text credential)  
+  [GitHub token](https://github.com/settings/tokens) for Github CLI
+* IMAGE_REGISTRY_CREDENTIALS (username/password credential)  
+  Credential to push image to the container registry (should have rights to `IMAGE_NAMESPACE`)
+* KOGITO_CI_EMAIL_TO (secret text credential)  
+  Email for notifications. You can set your email for example
```
Comment (Kevin-Mok) [inline README.md]:
```suggestion
* `GIT_AUTHOR_CREDS_ID` (username/password credential)  
  * Username / [GitHub token](https://github.com/settings/tokens) which has rights on `GIT_AUTHOR`
* `BOT_AUTHOR_CREDS_ID` (username/password credential)  
  * Username / [GitHub token](https://github.com/settings/tokens) which has rights on `BOT_AUTHOR`
* `GITHUB_TOKEN_CREDS_ID` (secret text credential)  
  * [GitHub token](https://github.com/settings/tokens) for Github CLI
* `IMAGE_REGISTRY_CREDENTIALS` (username/password credential)  
  * Credential to push image to the container registry (should have rights to `IMAGE_NAMESPACE`)
* `KOGITO_CI_EMAIL_TO` (secret text credential)  
  * Email for notifications. You can set your email for example
```
Same here.

**Review by Kevin-Mok — COMMENTED** [review]
Minor stylistic changes for README.

**Comment by radtriste** [comment]
waiting for dependent PRs to merge

**Review by sutaakar — APPROVED** [review]

**Review by Kevin-Mok — APPROVED** [review]

**Pull request merged**